plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Core/Type.hs

```diff
@@ -165,3 +171,47 @@ erase (TPLC.Error ann _)            = Error ann
 -- | Erase a Typed Plutus Core Program to its untyped counterpart.
 eraseProgram :: TPLC.Program tyname name uni fun ann -> Program name uni fun ann
 eraseProgram (TPLC.Program a v t) = Program a v $ erase t
+
+{-|
+A specialized version of 'Term' for evaluation. The 'name' and 'a' parameters are specialized to 'Name' ('Unique') and '()' (omitted).
+-}
+data ETerm uni fun
+    = EConstant !(Some (ValueOf uni))
+    | EBuiltin !fun
+    | EVar !TPLC.Unique
+    | ELamAbs !TPLC.Unique !(ETerm uni fun)
+    | EApply !(ETerm uni fun) !(ETerm uni fun)
+    | EDelay !(ETerm uni fun)
+    | EForce !(ETerm uni fun)
+    | EError
+    deriving stock (Show, Functor, Generic)
+    deriving anyclass (NFData)
+
+termToETerm :: Term TPLC.Name uni fun () -> ETerm uni fun
```

Comment:
why not more generalized argument: `Term Name uni fun a`?

Reply:
Because that throws away information silently, this does not. Here you have to explicitly opt-in to throwing away any annotations yourself before calling this.

More generally: `ETerm uni fun === Term Name uni fun ()`, and these functions witness that.